modules/xfeatures2d/src/boostdesc.cpp

```diff
@@ -0,0 +1,733 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ * Copyright (c) 2013, 2016
+ *
+ * Tomasz Trzcinski <t dot trzcinski at ii dot pw dot edu dot pl>
+ * Mario Christoudias <mariochristoudias at gmail dot com>
+ * Vincent Lepetit <lepetit at icg dot tugraz dot at>
+ *
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holders nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/*
+
+ "Learning Image Descriptors with Boosting"
+ T. Trzcinski, M. Christoudias and V. Lepetit
+ IEEE Transactions on Pattern Analysis and Machine Intelligence, 2013
+
+ "Boosting Binary Keypoint Descriptors"
+ T. Trzcinski, M. Christoudias, V. Lepetit and P. Fua
+ Computer Vision and Pattern Recognition (CVPR), 2013
+
+ Original code: Tomasz Trzcinski <t dot trzcinski at ii dot pw dot edu dot pl>
+ OpenCV port by: Cristian Balint <cristian dot balint at gmail dot com>
+
+ */
+
+#include <bitset>
+#include "precomp.hpp"
+
+#include "opencv2/imgproc/imgproc_c.h"
```

Comment:
You should avoid usage of the legacy interface without strong reasons. Just remove CV_ prefixes in L370 to use new enums.